src/hotkey/command_executor.cpp

```diff
@@ -516,10 +516,13 @@ static void event_execute( const SDL_Event& event, command_executor* executor)
 		return;
 	}
 
+	const hotkey_command& command = get_hotkey_command(hk->get_command());
 	bool press = event.type == SDL_KEYDOWN ||
 			event.type == SDL_JOYBUTTONDOWN ||
-			event.type == SDL_MOUSEBUTTONDOWN ||
-			event.type == SDL_TEXTINPUT;
+			event.type == SDL_MOUSEBUTTONDOWN;
+	if(!command.toggle) {
```

Comment:
What is this part for? "If the hotkey is a toggle hotkey, then SDL_TEXTINPUT events are always release events"?

Reply:
Yeah, basically toggle hotkeys don't handle TEXTINPUT at all.

Reply:
But it isn't what the `press` variable determines. `press` merely records whether the event was a key press or a key release.

Reply:
Technically, TEXTINPUT is neither a press nor a release in the first place; it was merely a convention that it be treated as a press. Still, maybe I could write this a little differently... in fact it may be entirely superfluous if the other parts ensure that we never actually reach this place under the required conditions.